0016-pre-shared-key/0016-pre-shared-key.md

```diff
@@ -0,0 +1,82 @@
+request Pre-Shared Key
+
+The Pre-Shared Key (PSK) protocol is an end-to-end transport protocol, used by the sender and receiver of an ILP payment to decide on a condition and fulfillment for a payment. The protocol also encrypts any additional data sent along with the payment.
+
+As the name suggests, PSK relies on a pre-shared secret key that is shared beforehand. In application-layer protocols like SPSP, the receiver generates the shared key and shares it with the sender over an end-to-end secure communication channel such as HTTPS. 
+
+An advantage of PSK is that the pre-shared secret key only needs to be shared once. So long as both parties possess this key and are listening for transfers, they can send payments between one another.
+
+A disadvantage of PSK is that it is repudiable. Knowledge of a payment's fulfillment cannot be used as cryptographic proof that the receiver completed the payment, because both sender and receiver know how to generate the fulfillment with their pre-shared secret key.
```

Comment:
Well no matter what, the sender needs to generate the condition, and therefore needs to know the fulfillment. If the receiver was the only one that knew how to generate the condition, there would have to be end-to-end communication on every payment.

Exchanging receiver and sender public keys and then generating a Diffie-Hellman shared secret could still be a useful way to get the pre-shared key though.